src/gui/redux/docgen/docgen.js

```diff
@@ -1,25 +1,3 @@
-import {
-  REQUEST_DOCGEN,
-  RECEIVED_DOCGEN,
-  RECEIVED_EMPTY_DOCGEN,
-  FAILED_DOCGEN,
-} from './docgen.actions'
+import factory from 'trampss-redux-factory'
 
-export const initialState = {
-  loaded: false,
-  data: {},
-}
-
-export default (state = initialState, { type, payload } = {}) => {
-  switch (type) {
-    case REQUEST_DOCGEN:
-      return { ...initialState, loaded: false }
-    case RECEIVED_DOCGEN:
-      return { data: payload, loaded: true }
-    case RECEIVED_EMPTY_DOCGEN:
-    case FAILED_DOCGEN:
-      return { ...initialState, loaded: true }
-    default:
-      return state
-  }
-}
+export default factory()()()({ name: 'docgen', type: 'uniq' })
```

Comment:
Not a really clear method signature.
It would be better to have the ability to write something like that : 
```es6
factory({ name: 'docgen', type: 'uniq' })
```

So we could have the possibility to write : 
- with curry functions (like today)
- all configuration in an object (like the above example).